pulsar-client-cpp/lib/ClientConnectionContainer.h

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef PULSAR_CPP_CLIENTCONNECTIONCONTAINER_H
+#define PULSAR_CPP_CLIENTCONNECTIONCONTAINER_H
+
+#include <vector>
+#include <iostream>
+#include <algorithm>    // std::min
+namespace pulsar {
+/* @brief - This class uses a vector to store elements and provides a wrap around getNext() function to retrieve elements in a round robin fashion.
+ * @note - this class is not thread safe.
+ */
+template<class T>
+class ClientConnectionContainer {
+ private:
+    size_t capacity_;
+    size_t currentIndex_;
+    std::vector<T> list_;
```

Comment:
Changed the name to array, I am using a vector as a fixed size array since boost::array requires the size to be specified during compile time.

Using class template since we can reuse the round robin array in the future - I didn't find a standard implementation for a cyclic iterator the closest I got was a cyclic buffer but that too wasn't good enough.